### DISCUSSION TITLE
How to add touch buttons?

### DISCUSSION THREAD

**Discussion by denizsahin7744** [HomeSpan/HomeSpan#766] [Q&A] [answered]
I want to add couple of ttp223b touch sensor to my project. When i looked examples i didn't understand how to integrate sensors to code because i only want to control state of relays. Can you guys help me to do that ?

Board = Doit ESP32 Devkit V1
Program i use = Platformio

main.cpp
```
#include "HomeSpan.h" 
#include "DEV_RELAY.h"

const int relayPin1 = 25;
const int relayPin2 = 26;

void setup() {
  Serial.begin(115200);

  homeSpan.setPairingCode("11122333");
  homeSpan.setQRID("111-22-333");

  homeSpan.begin(Category::Bridges, "HomeSpan Bridge");
       
  new SpanAccessory();
    new Service::AccessoryInformation();
      new Characteristic::Identify();
      new SpanButton(12);
    new DEV_RELAY(relayPin1);


  new SpanAccessory();
    new Service::AccessoryInformation();
      new Characteristic::Identify();
    new DEV_RELAY(relayPin2);

}

void loop() {
  homeSpan.poll();
}
```

DEV_RELAY.h
```
struct DEV_RELAY : Service::Switch {  // deriving from the HomeSpan Switch service

  int relayPin;
  SpanCharacteristic *power;

  DEV_RELAY(int relayPin) : Service::Switch() {

    power = new Characteristic::On();
    this->relayPin = relayPin;
    pinMode(relayPin, OUTPUT);
  }

  boolean update() {
    digitalWrite(relayPin, power->getNewVal());
   
    return(true);
  }
};
```

**Comment by HomeSpan** [comment, marked as answer]
Start with using regular push buttons and work your way through the tutorials that show you explicitly how to use them.  Then, after everything is working replace your push buttons with touch sensors based on a very slight modification to the code as described in the touch-sensor section of the tutorial.

**Reply by denizsahin7744** [reply]
okey i will try it thank you

**Comment by HomeSpan** [comment]
Also, I noticed that your use of setQRID is incorrect.  Please review the HomeSpan documentation on QR Codes to make sure you are using it correctly.  Note that if you do not intend on creating your own QR Code label, there is no reason to call this method.

**Reply by denizsahin7744** [reply]
thank you for noticing
